dev/validate_bom__config.py

```diff
@@ -1645,6 +1645,67 @@ def add_files_to_upload(self, options, file_set):
     """Implements interface."""
     pass
 
+class PluginStageConfigurator(Configurator):
+  """Controls hal plugins."""
+
+  def init_argument_parser(self, parser, defaults):
+    """Implements interface."""
+    add_parser_argument(
+        parser, 'plugin_repo', defaults, None,
+        help='The address of the plugin repository to pull metadata from.')
+
+    add_parser_argument(
+        parser, 'plugin_name', defaults, None,
+        help='The name of the plugin to deploy.')
+
+    add_parser_argument(
+        parser, 'plugin_extension', defaults, None,
+        help='The name of the plugin extension to deploy.')
+
+    add_parser_argument(
+        parser, 'plugin_version', defaults, None,
+        help='The plugin version to deploy.')
+
+  def add_config(self, options, script):
+    """Implements interface."""
+    if options.plugin_repo is None:
+      return
+
+    script.append(
+        'hal -q --log=info plugins repository add spin-test-repo'
+        ' --url={repo}'.format(repo=options.plugin_repo))
+
+    script.append(
+        'hal -q --log=info plugins add {name}'
```

Comment:
One thing that worries me a bit here is that we're loading the plugins for *all* tests, so if there's some issue with that particular plugin or if it breaks in the future the deploy will fail as opposed to just the plugin test.

That being said, by design we only deploy once as it takes a long time; and I guess the version is pinned, right, so it won't randomly break?

Reply:
That's a fair thing to worry about and probably worth thinking about. This plugin is very narrowly scoped but other plugins might have implications for other integration tests (e.g., a custom clouddriver credential loader) or even implications for other plugins. This probably isn't the right way to test individual plugins.

Yes re: the plugin version being pinned. Loading plugins could break the whole test suite if there's some bug in the Kork plugin-loader code that prevents a service from starting up. But that kind of bug exists for any feature that's enabled that happens to bring down a Spinnaker service...

Reply:
Yeah, I think this makes sense then given that it's a super simple pinned plugin that is just there to ensure that we can deploy with a plugin.